spyder/widgets/sourcecode/codeeditor.py

```diff
@@ -1201,6 +1207,15 @@ def set_blanks_enabled(self, state):
         # Rehighlight to make the spaces less apparent.
         self.rehighlight()
 
+    def set_scrollpastend_enabled(self, state):
+        """
+        Allow user to scroll past the end of the document to have the last
+        line on top of the screen
+        """
+        self.scrollpastend_enabled = state
+        self.setCenterOnScroll(state)
+        self.update()

```

Comment:
replace `self.update()` by `self.setDocument(self.document())` and it should solve your issue about the option not being applied at runtime.

Reply:
Works like a charm! Thanks @jnsebgosselin!